modules/rostests/apitests/gdi32/TextTransform.c

```diff
@@ -302,27 +447,79 @@ static void DoTestEntry(const TEST_ENTRY *entry, HDC hDC, HBITMAP hbm)
 
         SetTextAlign(hDC, entry->TextAlign);
 
-        TextOutW(hDC, entry->ptRef.x, entry->ptRef.y, &s_chBlackBox, 1);
+        TextOutW(hDC, entry->ptRef.x, entry->ptRef.y, s_szBlackBoxes, lstrlenW(s_szBlackBoxes));
 
         ModifyWorldTransform(hDC, NULL, MWT_IDENTITY);
 
-        for (i = 0; i < entry->cWhitePoints; ++i)
-        {
-            rgb = GetPixel(hDC, entry->WhitePoints[i].x, entry->WhitePoints[i].y);
-            ok(rgb == WHITE, "Line %d: %d: (%ld, %ld) is not white\n", entry->line, i,
-               entry->WhitePoints[i].x, entry->WhitePoints[i].y);
-        }
+#define POS(ix, iy) {(ix) * WIDTH/4 + WIDTH/8, (iy) * HEIGHT/4 + HEIGHT/8}
 
-        for (i = 0; i < entry->cBlackPoints; ++i)
+        for (j = 0; j < 4; ++j)
         {
-            rgb = GetPixel(hDC, entry->BlackPoints[i].x, entry->BlackPoints[i].y);
-            ok(rgb == BLACK, "Line %d: %d: (%ld, %ld) is not black\n", entry->line, i,
-               entry->BlackPoints[i].x, entry->BlackPoints[i].y);
+            for (i = 0; i < 4; ++i)
+            {
+                POINT pt = POS(i, j);
+                INT k = j * 4 + i + 1;
+                COLORREF rgb = GetPixel(hDC, pt.x, pt.y);
+                BOOL bFound = FALSE;
+                INT m;
+                for (m = 0; m < _countof(entry->aBlacks); ++m)
+                {
+                    if (entry->aBlacks[m] == 0)
+                        break;
+                    if (entry->aBlacks[m] == k)
+                    {
+                        bFound = TRUE;
+                        break;
+                    }
+                }
+
+#if 1
```

Comment:
debugging leftover? or should it become `#ifdef DEBUGGING` ?

Reply:
Deleted.